rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -148,11 +148,12 @@ public boolean save_state(OutputObjectState os, int ot) {
                 os.packString(status.name());
             }
 
-            if (afterLRAListeners == null) {
+            if (endLRAListeners == null) {
                 os.packInt(0);
             } else {
-                os.packInt(afterLRAListeners.size());
-                for (AfterLRAListener participant : afterLRAListeners) {
+                os.packInt(endLRAListeners.size());
+                for (EndLRAListener participant : endLRAListeners) {
```

Comment:
Doesn't the LRARecord pack these already: https://github.com/mmusgrov/narayana/blob/LRA/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java#L679 - although I am not then sure why the current LRA branch code also packs the AfterLRA elements: https://github.com/jbosstm/narayana/blob/LRA/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java#L156

Reply:
yes, afterLRAlisteners even now in the master are handled in the Transaction class, not in LRARecord. My solution for forgetting nested completed participants (this PR) is building on top of it which is the reason for the refactoring.

@

Reply:
Thanks @xstefank it's not strictly related to this PR but I would like to know why there is save/restore of the AfterLRA in LRARecord and Transaction /cc @mmusgrov

Reply:
there might be even a bigger issue with this approach (afterLRA listeners in Transaction) which would, unfortunately, invalidate even this PR :). I am working on a test case that will help.

Reply:
@tomjenkinson @mmusgrov FYI I created https://issues.redhat.com/browse/JBTM-3239. Let's continue the discussion in the issue, please.